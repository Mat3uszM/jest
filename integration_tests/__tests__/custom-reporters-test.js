/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */
const runJest = require('../runJest');
const skipOnWindows = require('skipOnWindows');

describe('Custom Reporters', () => {
  skipOnWindows.suite();
  
  test('TestReporter with all tests passing', () => {
    const {
      stdout,
      status,
      stderr,
    } = runJest('custom_reporters', ['add-test.js']);
    let parsedJSON;

    try {
      parsedJSON = JSON.parse(stdout);
    } catch (error) {
      throw new Error(
        'Failed to parse JSON, Check the Output of TestReporter'
      );
    }

    const {onRunComplete, onRunStart, onTestResult, onTestStart} = parsedJSON;

    expect(status).toBe(0);
    expect(stderr.trim()).toBeFalsy();

    expect(onRunComplete.numPassedTests).toBe(1);
    expect(onRunComplete.numFailedTests).toBe(0);
    expect(onRunComplete.numTotalTests).toBe(1);

    expect(onRunStart.called).toBeTruthy();
    expect(onTestResult.called).toBeTruthy();
    expect(onTestStart.called).toBeTruthy();

    expect(parsedJSON).toMatchSnapshot();
  });

  test('TestReporter with all tests failing', () => {
    let parsedJSON;
    const {
      stdout,
      status,
      stderr,
    } = runJest('custom_reporters', ['add-fail-test.js']);
    
    try {
      parsedJSON = JSON.parse(stdout);
    } catch (error) {
      throw new Error('Failed to parse JSON. Check the output of TestReporter');
    }

    const {onTestStart, onTestResult, onRunStart, onRunComplete} = parsedJSON;

    expect(status).toBe(1);
    expect(stderr).toBeFalsy();

    expect(onRunComplete.numPassedTests).toBe(0);
    expect(onRunComplete.numFailedTests).toBe(1);
    expect(onRunComplete.numTotalTests).toBe(1);

    expect(onRunStart.called).toBeTruthy();
    expect(onTestStart.called).toBeTruthy();
    expect(onTestResult.called).toBeTruthy();

    expect(parsedJSON).toMatchSnapshot();
  });

  test('IncompleteReporter for flexibility', () => {
    const {stdout, status} = runJest('custom_reporters', [
      '--config',
      JSON.stringify({
        'reporters': [
          '<rootDir>/reporters/IncompleteReporter.js',
        ],
      }),
      'add-test.js',
    ]);

    expect(status).toBe(0);

    expect(stdout).toMatch('onRunComplete is called');
    expect(stdout).toMatch('Passed Tests: 1');
    expect(stdout).toMatch('Failed Tests: 0');
    expect(stdout).toMatch('Total Tests: 1');

    expect(stdout).toMatchSnapshot();
  });
});
