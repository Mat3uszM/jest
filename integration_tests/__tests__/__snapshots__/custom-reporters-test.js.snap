// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Reporters Integration IncompleteReporter for flexibility 1`] = `
"onRunComplete is called
Passed Tests: 1
Failed Tests: 0
Total Tests: 1
"
`;

exports[`Custom Reporters Integration TestReporter with all tests failing 1`] = `
Object {
  "onRunComplete": Object {
    "called": true,
    "config": "object",
    "numFailedTests": 1,
    "numPassedTests": 0,
    "numTotalTests": 1,
  },
  "onRunStart": Object {
    "called": true,
    "config": "object",
    "options": "object",
  },
  "onTestResult": Object {
    "called": true,
    "times": 1,
  },
  "onTestStart": Object {
    "called": true,
    "config": false,
    "path": true,
  },
  "options": Object {
    "option1": "Hello",
    "option2": "World",
    "option3": "Amazing!!!",
  },
}
`;

exports[`Custom Reporters Integration TestReporter with all tests passing 1`] = `
Object {
  "onRunComplete": Object {
    "called": true,
    "config": "object",
    "numFailedTests": 0,
    "numPassedTests": 1,
    "numTotalTests": 1,
  },
  "onRunStart": Object {
    "called": true,
    "config": "object",
    "options": "object",
  },
  "onTestResult": Object {
    "called": true,
    "times": 1,
  },
  "onTestStart": Object {
    "called": true,
    "config": false,
    "path": true,
  },
  "options": Object {
    "option1": "Hello",
    "option2": "World",
    "option3": "Amazing!!!",
  },
}
`;
